tests/nosetests/pyanaconda_tests/localization_test.py

```diff
@@ -17,72 +17,214 @@
+
+    def locale_timezones_test(self):
+        self.assertIn("Europe/Oslo", localization.get_locale_timezones("no"))
+
+    @patch.dict("pyanaconda.localization.os.environ", dict())
+    def xlated_tz_test(self):
+        localization.os.environ["LANG"] = "en_US"
+        self.assertEqual("Europe/Barcelona", localization.get_xlated_timezone("Europe/Barcelona"))
+        localization.os.environ["LANG"] = "cs_CZ"
+        self.assertEqual("Evropa/Praha", localization.get_xlated_timezone("Europe/Prague"))
+        localization.os.environ["LANG"] = "blah"
+        self.assertRaises(localization.InvalidLocaleSpec,
+                          localization.get_xlated_timezone,
+                          "America/New_York")
+
+
+class SetupLocaleTest(unittest.TestCase):
+
+    @patch("pyanaconda.localization.setenv")
+    @patch("pyanaconda.localization.locale_mod.setlocale")
+    @patch("pyanaconda.localization.set_modules_locale")
+    def setup_locale_notext_test(self, set_modules_locale_mock, setlocale_mock, setenv_mock):
+        """Test setup_locale in GUI mode"""
+
+        loc_proxy = MagicMock()
+
+        locale = localization.setup_locale("sk", localization_proxy=loc_proxy)
+
+        loc_proxy.SetLanguage.assert_called_once_with("sk")
+        setenv_mock.assert_called_once_with("LANG", "sk")
+        setlocale_mock.assert_called_once_with(locale_mod.LC_ALL, "sk")
+        set_modules_locale_mock.assert_called_once_with("sk")
+
+        self.assertEqual(locale, "sk")
+
+    @patch.dict("pyanaconda.localization.os.environ", dict())
+    @patch("pyanaconda.localization.locale_supported_in_console", return_value=False)
+    @patch("pyanaconda.localization.setenv")
+    @patch("pyanaconda.localization.locale_mod.setlocale")
+    @patch("pyanaconda.localization.set_modules_locale")
+    def setup_locale_text_test(self, set_modules_locale_mock, setlocale_mock, setenv_mock,
+                               locale_supported_in_console_mock):
+        """Test setup_locale in TUI mode"""
+        # note: to eliminate unpredictable support in console, mocking such that it always fails
+
+        locale = localization.setup_locale("ja_JP", text_mode=True)
+
+        locale_supported_in_console_mock.assert_called_once_with("ja_JP")
+        self.assertEqual(localization.os.environ["LANG"], DEFAULT_LANG)
+        setenv_mock.assert_called_once_with("LANG", DEFAULT_LANG)
+        setlocale_mock.assert_called_once_with(locale_mod.LC_ALL, DEFAULT_LANG)
+        set_modules_locale_mock.assert_called_once_with(DEFAULT_LANG)
+
+        self.assertEqual(locale, DEFAULT_LANG)
+
+    @patch("pyanaconda.localization.setenv")
+    @patch("pyanaconda.localization.locale_mod.setlocale", side_effect=[locale_mod.Error, None])
+    @patch("pyanaconda.localization.set_modules_locale")
+    def setup_locale_setlocale_fail_test(self, set_modules_locale_mock, setlocale_mock, setenv_mock):
+        """Test setup_locale with failure in setlocale"""
+
```

Comment:
Please make the newlines on a function/method start consistent. I saw three types here.

1:
```
def test():
"""documentation

description
"""
   code
```

2:
```
def test():
"""documentation

description
"""

   code
```

3:
```
def test():
"""documentation

description

"""
   code
```


You should be consistent with [PEP-257](https://www.python.org/dev/peps/pep-0257/) which means the first variant.